client/galaxy/scripts/mvc/tool/tool-form-composite.js

```diff
@@ -456,7 +456,8 @@ define([ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view'
                         if ($.isArray( response ) && response.length > 0) {
                             self.$el.append( $( '<div/>', { id: 'webhook-view' } ) );
                             var WebhookApp = new Webhooks.WebhookView({
-                                urlRoot: Galaxy.root + 'api/webhooks/workflow'
+                                urlRoot: Galaxy.root + 'api/webhooks/workflow',
+                                toolId: job_def.tool_id
                             });
```

Comment:
At an initial glance - it seems like version should be required here as well - if there are multiple tools with the same id but different versions loaded. In the tool shed case they will have separate IDs - but if they are locally added tools I feel like we should handle that case.